docs/references/tanstack-start/read-session-data.mdx

```diff
@@ -0,0 +1,142 @@
+---
+title: Read session and user data in your TanStack Start app with Clerk
+description: Learn how to use Clerk's hooks and helpers to access the active session and user data in your TanStack Start application.
+---
+
+Clerk provides a set of [hooks and helpers](/docs/references/tanstack-start/overview#client-side-helpers) that you can use to access the active session and user data in your TanStack Start application. Here are examples of how to use these helpers in both the client and server side to get you started.
+
+## Server side
+
+### `getAuth()`
+
+To use `getAuth` on the server side, you can import it from `@clerk/tanstack-start/server` and use it in your server functions and API Routes to access the `Auth` object.
+
+<Tabs type="function" items={["Before load Function", "API Endpint"]}>
+  <Tab>
+    ```tsx {{ filename: 'app/routes/example.tsx' }}
+    import { getAuth } from '@clerk/tanstack-start/server'
+    import { createFileRoute, redirect } from '@tanstack/react-router'
+    import { createServerFn } from '@tanstack/start'
+
+    const getAuthenticatedUser = createServerFn('POST', async (_, { request }) => {
+      const auth = await getAuth(request)
+
+      if (!auth.userId) {
+        throw redirect({
+          to: '/sign-in',
+        })
+      }
+
+      return { auth }
+    })
+
+    export const Route = createFileRoute('/example')({
+      component: Example,
+      beforeLoader: async () => {
+        const { auth } = await getAuthenticatedUser()
+
+        return { auth }
+      },
+      loader: async ({ context }) => {
+        return {
+          userId: context.auth.userId,
+        }
+      },
+    })
+
+    async function Example() {
+      const { userId } = await getAuth()
```

Comment:
`const state = Route.useLoaderData()` ?
if you are doing `const { userId } = await getAuth()` , then why need to create `getAuthenticatedUser` and pass it to `loader` ?

can you explain this example to me

Reply:
Fixed!